File: data structure in c/binarySearch.c
#include<stdio.h>

int binarySearch(int arr[] , int n , int key){
    int l = 0 , r = n - 1;
    while(l <= r){
        int m = l + (r - l)/ 2;
        if (arr[m] == key){
            return m;
        }
        if (arr[m] < key){
            l = m + 1;
        }
        if (arr[m] > key){
            r = m - 1;
        }       
    }
    return -1;
}

void main(){
    int arr[] = {10 , 20 , 30 , 40 , 50 , 60 , 70 , 80 , 90 , 100};
    int n = sizeof(arr) / sizeof(arr[0]);
    int tar = 50;
    int result = binarySearch(arr , n , tar);
    if (result != -1){
        printf("Target found at location/index : %d\n", result);
    }
    else{
        printf("Target not found exit code %d\n", result);
        
    }
}

File: python/binarySearch.py
def binarySearch(arr, key, l, r):
    if l > r:
        return -1
    mid = (l + r) // 2
    if arr[mid] == key:
        return mid
    elif arr[mid] > key:
        return binarySearch(arr, key, l, mid - 1)
    else:
        return binarySearch(arr, key, mid + 1, r)


array = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
key = int(input("Enter key between 1 to 9 :"))
left = 0
right = len(array) - 1
index = binarySearch(array, key, left, right)
if index == -1:
    print("key", key, "is not found")
else:
    print("key", key, "is found at index", index)
